c-pallets/file-map/src/lib.rs

```diff
@@ -127,12 +138,65 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn bond_acc)]
 	pub(super) type BondAcc<T: Config> =
+	// Same as for SchedulerMap. Wrong use of StringLimit. Needs new constant
 		StorageValue<_, BoundedVec<AccountOf<T>, T::StringLimit>, ValueQuery>;
 
 	#[pallet::pallet]
 	#[pallet::generate_store(pub(super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberOf<T>> for Pallet<T> {
+		//Polling exception report
+		fn on_initialize(now: BlockNumberOf<T>) -> Weight {
+			let number: u128 = now.saturated_into();
+			let count: usize = Self::scheduler_map().len();
+			if number % 1200 == 0 {
+				let alregister_list = SchedulerMap::<T>::get();
+				let mut alregister_controller_list: BoundedVec<AccountOf<T>, T::StringLimit> =
+					Default::default();
+				for alregister in alregister_list.iter() {
+					if let Err(e) = alregister_controller_list
+						.try_push(alregister.controller_user.clone())
+						.map_err(|_| Error::<T>::StorageLimitReached)
+					{
+						log::error!("FileMap error: {:?}", e);
+					}
+				}
+
+				for v in BondAcc::<T>::get().iter() {
+					if !alregister_controller_list.contains(&v) {
+						for v2 in alregister_list.iter() {
+							if v2.controller_user == v.clone() {
+								pallet_cess_staking::slashing::slash_scheduler::<T>(&v2.stash_user);
+								T::CreditCounter::record_punishment(&v2.controller_user);
+							}
+						}
+					}
+				}
+
+				let mut ctl: BoundedVec<AccountOf<T>, T::StringLimit> = Default::default();
+				for (_stash, controller) in pallet_cess_staking::Bonded::<T>::iter() {
+					if let Err(e) = ctl
+						.try_push(controller.clone())
+						.map_err(|_| Error::<T>::StorageLimitReached)
+					{
+						log::error!("FileMap error: {:?}", e);
+					}
+				}
+				BondAcc::<T>::put(ctl);
+				for (key, value) in <SchedulerException<T>>::iter() {
+					if value.count > (count / 2) as u32 {
+						pallet_cess_staking::slashing::slash_scheduler::<T>(&key);
+					}
+
+					<SchedulerException<T>>::remove(key);
+				}
+			}
+			0 // Shouldn't be 0 | fix
```

Comment:
At present, the logic here has been discarded, so I deleted it in the new version